app/Http/Controllers/CommunityController.php

```diff
@@ -40,8 +45,36 @@ public function getChat()
         return view('community.chat');
     }
 
-    public function getLive()
+    public function getLive(Request $request)
     {
-        return view('community.live');
+        $streams = null;
+        $featuredStream = null;
+        if (!Cache::has("livestreams"))
+        {
+            $justin_api_url = "https://api.twitch.tv/kraken/streams?on_site=1&limit=40&offset=0&game=Osu!";
+            $data = json_decode(file_get_contents($justin_api_url));
+            $streams = $data->streams;
+            Cache::put('livestreams', $streams, 300);
+        } else {
+            $streams = Cache::get("livestreams");
+        }
+        
+        if (Auth::user() != null && Auth::user()->isGmt()) { 
+            if ($request->has("promote")) 
```

Comment:
although, to make a form, I need to add onclick submit jquery code for <a> element? or am I doing something wrong?